components/brave_extension/extension/brave_extension/background/events/cosmeticFilterEvents.ts

```diff
@@ -37,13 +37,15 @@ chrome.contextMenus.onClicked.addListener(function (info, tab) {
     case 'addBlockElement': {
       chrome.tabs.query({ active: true, currentWindow: true }, function (tabs: any) {
         chrome.tabs.sendMessage(tabs[0].id, { type: 'getTargetSelector' }, function (response: any) {
-          if (response) {
-            rule.selector = window.prompt('CSS selector to block: ', `${response}`) || ''
-            chrome.tabs.insertCSS({
-              code: `${rule.selector} {display: none;}`
-            })
-            cosmeticFilterActions.siteCosmeticFilterAdded(rule.host, rule.selector)
+          if (!response) {
+            rule.selector = window.prompt('We were unable to automatically populate a correct CSS selector for you. Please manually enter a CSS selector to block:') || ''
+          } else {
+            rule.selector = window.prompt('CSS selector:', `${response}`) || ''
           }
+          chrome.tabs.insertCSS({
```

Comment:
Should we only insert the rule if a valid css selector resulted from the prompt? Perhaps at least check for a non-empty string

Reply:
We want to give the user an opportunity to insert their own CSS selector if they want - otherwise they're forced to right click on something that the selector thinks is valid instead of the element they want to block.

Reply:
@Snuupy that makes sense - but I think @petemill is asking: what happens if the user deletes the selector (or provides an invalid selector)

Reply:
Especially for the version that states a selector could not be found and to enter one from scratch. What if the user cancels or enters a blank selector?

Reply:
If the user cancels, nothing is inserted. Currently, if a user enters a blank selector, a blank selector is added, and is attempted to be applied, but nothing ends up happening because an empty selector always returns nothing. If an invalid selector is entered, unfortunately right now (until the filter list import/export/editing functionality is completed) they have to clear all rules for the website to try again.

Reply:
Updated to not insert if selector is blank/falsey 👍